### PR TITLE
fix: GitHub app admin install

### PR DESCRIPTION
**Pull request by rephus**
Closes:  https://github.com/lightdash/lightdash/issues/11987

### Description:

image

CLicking on pending approval takes it to the setup page, where they can cancel the request 

[Screencast from 21-10-24 10:57:41.webm](https://github.com/user-attachments/assets/713dfd7b-fae3-4d1a-963b-eb7f23cc028c)

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

File: packages/backend/src/services/GithubAppService/GithubAppService.ts [hunk 1]
```diff
@@ -111,6 +101,57 @@ export class GithubAppService extends BaseService {
...
+            const redirectUrl = new URL(oauth?.returnTo || '/');
+
+            if (setup_action === 'request') {
+                // User attempted to setup the app, didn't have permission in GitHub and sent a request to the admins
+                // We will try to poll for the installation id
+                console.info(
+                    'Waiting for Github app to be authorized by an admin',
+                );
+
+                const interval = setInterval(async () => {
```
Comment (rephus) [inline packages/backend/src/services/GithubAppService/GithubAppService.ts:113]: Please read this https://github.com/lightdash/lightdash/issues/11987#issuecomment-2425893582

**Comment by rephus** [comment]
@joaoviana this is a bit hard to replicate, so feel free to skip some of the testing. Or use some repo on Lightdash to test part of the flow.

File: packages/frontend/src/components/UserSettings/GithubSettingsPanel/index.tsx [hunk 2]
```diff
@@ -82,7 +89,34 @@ const GithubSettingsPanel: FC = () => {
     const { data, isError, isInitialLoading } = useGitHubRepositories();
     const deleteGithubInstallationMutation =
         useDeleteGithubInstallationMutation();
+
+    const status = useSearchParams('status');
+    const { showToastWarning } = useToaster();
+    const isWaitingForGithubRequest = status === 'github_request_sent';
```
Comment (joaoviana) [inline packages/frontend/src/components/UserSettings/GithubSettingsPanel/index.tsx:95]: does the search param clear after installation?

**Review by joaoviana — APPROVED** [review]

File: packages/frontend/src/components/UserSettings/GithubSettingsPanel/index.tsx (hunk 2 above)
Comment (rephus) [reply packages/frontend/src/components/UserSettings/GithubSettingsPanel/index.tsx:95]: nope, but it is ignored if the request to github returns the right results

**Pull request merged**